Standards/scs-XXXX-v1-volume-backup-service.md

```diff
@@ -0,0 +1,89 @@
+
+This standard aims to make this functionality the default in SCS clouds so that customers can expect the feature to be usable.
+
+## Design Considerations
+
+The standard should make sure that the feature is available and usable but should not limit the exact implementation (e.g. choice of backend driver) any further than necessary.
+
+### Options considered
+
+#### Only recommend volume backup feature, use images as alternative
+
+As an alternative to the volume backup feature of the Block Storage API, Glance images can also be created based on volumes and act as a backup under certain circumstances.
+As an option, this standard could keep the actual integration of the volume backup feature optional and guide users how to use images as backup targets instead in case the feature is unavailable.
+
+However, it is not guaranteed that the Glance backend storage is separate from the volume storage.
+For instance, both could be using the same Ceph cluster.
+In such case, the images would not count as genuine backups.
+
+Although users are able to download images and transfer them to a different storage location, this approach might also prove unfeasible depending on the image size and the existence (or lack) of appropriate target storage on the user side.
+
+Furthermore, incremental backups are not possible when creating Glance images from volumes either.
+This results in time-consuming backup operations of fully copying a volume everytime a backup is created.
+
+#### Focus on feature availability, make feature mandatory
+
+This option is pretty straightforward.
+It would make the volume backup feature mandatory for SCS clouds.
+This way users can expect the feature to be available and usable.
+
+With this, users can leverage functionalities like incremental backups and benefit from optimized performance of the backup process due to the tight integration with the volume service.
+
+However, it does not seem feasible to also mandate having a separate storage backend for volume backups at the same time due to potential infrastructure limitations at CSP-side making it hard or even impossible to offer.
+As such, the actual benefit of backups in terms of reliability and security aspects would be questionable if a separate storage backend is not mandated and therefore not guaranteed.
+
+This approach would focus on feature availability rather than backup reliability.
+
+#### Focus on backup reliability, make separate backend mandatory
+
+As an alternative, the volume backup feature availability could be made optional but in case a CSP chooses to offer it, the standard would mandate a separate storage backend to be used for volume backups.
+This way, failures of the volume storage backend would not directly impact the availability and safety of volume backups, making them actually live up to their name.
+
+In contrast to the above, this approach would focus on backup reliability rather than feature availability.
+
```

Comment:
Speaking from a users perspective: I want to have a backup, that I can definitely use, when the original is not there anymore.
For volumes this would mean, that the volume backup should be in a different backend, in case of a backend failure or even an AZ outage.

On the contrary this could lead to deployments with only one Ceph backend, which spans over three fire zones and is configured with internal replication between those zones, would not be allowed to offer the volume backup feature.

I think we should discuss this with CSPs and in the IaaS call.

Reply:
> I think we should discuss this with CSPs and in the IaaS call.

Agreed. Added it to the next IaaS call.